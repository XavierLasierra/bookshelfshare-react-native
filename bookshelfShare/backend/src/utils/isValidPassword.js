const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
