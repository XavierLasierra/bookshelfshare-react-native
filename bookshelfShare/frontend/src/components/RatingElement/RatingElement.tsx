/* eslint-disable no-underscore-dangle */
import React from 'react';
import { View, Text } from 'react-native';
import { AirbnbRating } from 'react-native-ratings';
import globalStyles from '../../styles/global.styles';

import styles from './ratingElement.styles';

export default function RatingElement({ rating, yours }: any) {
  return (
    <View style={styles.ratingContainer} key={`userRating-${rating.user._id}`}>
      <View style={styles.nameRatingContainer}>
        <Text style={globalStyles.titleText}>
          {rating.user.username}
          {' '}
          {yours && '(you)'}
        </Text>
        <AirbnbRating
          showRating={false}
          count={5}
          defaultRating={rating.rating}
          size={15}
          isDisabled
        />
      </View>
      <View style={styles.reviewContainer}>
        <Text style={styles.rating}>{rating.review}</Text>
      </View>
    </View>
  );
}
