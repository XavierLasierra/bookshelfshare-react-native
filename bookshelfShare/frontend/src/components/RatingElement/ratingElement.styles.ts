import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  ratingContainer: {
    backgroundColor: stylesConstants.colors.white,
    borderRadius: 5,
    marginVertical: 5,
    padding: stylesConstants.padding.small
  },
  nameRatingContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomWidth: 1,
    borderColor: stylesConstants.colors.secondary
  },
  reviewContainer: {
    paddingVertical: stylesConstants.padding.small
  },
  rating: {
    color: stylesConstants.colors.mainText,
    fontSize: stylesConstants.fontSize.mid
  }
});
