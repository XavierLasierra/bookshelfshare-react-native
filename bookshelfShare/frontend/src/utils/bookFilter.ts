export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
