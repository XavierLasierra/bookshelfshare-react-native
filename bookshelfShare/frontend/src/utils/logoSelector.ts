import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
