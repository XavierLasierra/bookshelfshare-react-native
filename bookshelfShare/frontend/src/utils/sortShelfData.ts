export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
