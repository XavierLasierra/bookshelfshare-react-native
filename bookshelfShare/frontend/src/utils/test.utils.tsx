import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
