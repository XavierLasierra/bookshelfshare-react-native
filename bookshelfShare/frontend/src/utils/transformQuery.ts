interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
