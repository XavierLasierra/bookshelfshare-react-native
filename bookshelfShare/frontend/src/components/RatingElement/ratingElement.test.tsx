import React from 'react';
import RatingElement from './RatingElement';
import {render} from '../../utils/test.utils';
import ratingsMock from '../../mocks/ratings.mock';

describe('Given an RatingElement component', () => {
  describe('When it is rendered', () => {
    describe('And the rating is not yours', () => {
      test('Then should match the snapshot', () => {
        const screen = render(
          <RatingElement rating={ratingsMock.ratings[0]} />,
        );
        expect(screen).toMatchSnapshot();
      });
    });
    describe('And the rating is yours', () => {
      test('Then should match the snapshot', () => {
        const screen = render(
          <RatingElement rating={ratingsMock.ratings[0]} yours />,
        );
        expect(screen).toMatchSnapshot();
      });
    });
  });
});
