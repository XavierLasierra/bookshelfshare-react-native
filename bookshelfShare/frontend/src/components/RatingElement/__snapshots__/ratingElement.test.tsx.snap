// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an RatingElement component When it is rendered And the rating is not yours Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "marginVertical": 5,
      "padding": 10,
    }
  }
  testID="ratingElement"
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ECECEC",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 5,
        }
      }
    >
      Xavi
       
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#BDC3C7",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#BDC3C7",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#BDC3C7",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 16,
        }
      }
    >
      meh
    </Text>
  </View>
</View>
`;

exports[`Given an RatingElement component When it is rendered And the rating is yours Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "marginVertical": 5,
      "padding": 10,
    }
  }
  testID="ratingElement"
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ECECEC",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 5,
        }
      }
    >
      Xavi
       
      (you)
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#BDC3C7",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#BDC3C7",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            collapsable={false}
            nativeID="animatedComponent"
            source={
              Object {
                "testUri": "../../react-native-ratings/dist/images/airbnb-star.png",
              }
            }
            style={
              Object {
                "height": 15,
                "margin": 3,
                "tintColor": "#BDC3C7",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 15,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3E3C37",
          "fontSize": 16,
        }
      }
    >
      meh
    </Text>
  </View>
</View>
`;
